tests/by-util/test_printf.rs

```diff
@@ -546,6 +569,68 @@ fn sub_any_asterisk_negative_first_param() {
         .stdout_only("a(x    )b"); // Would be 'a(    x)b' if -5 was 5
 }
 
+#[test]
+fn sub_any_asterisk_first_param_with_integer() {
+    new_ucmd!()
+        .args(&["|%*d|", "3", "0"])
+        .succeeds()
+        .stdout_only("|  0|");
+
+    new_ucmd!()
+        .args(&["|%*d|", "1", "0"])
+        .succeeds()
+        .stdout_only("|0|");
+
+    new_ucmd!()
+        .args(&["|%*d|", "0", "0"])
+        .succeeds()
+        .stdout_only("|0|");
+
+    new_ucmd!()
+        .args(&["|%*d|", "-1", "0"])
+        .succeeds()
+        .stdout_only("|0|");
+
+    new_ucmd!()
```

Comment:
Mind adding a comment here that explains that negative widths get aligned to the left? It's a bit tricky to see.

Reply:
Sure, just pushed (also bumped the width to 3 to make it a little more visible